v2/carbs/carbs_test.go

```diff
@@ -0,0 +1,80 @@
+package carbs
+
+import (
+	"os"
+	"testing"
+)
+
+/*
+func mkCar() (string, error) {
+	f, err := ioutil.TempFile(os.TempDir(), "car")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	ds := mockNodeGetter{
+		Nodes: make(map[cid.Cid]format.Node),
+	}
+	type linker struct {
+		Name  string
+		Links []*format.Link
+	}
+	cbornode.RegisterCborType(linker{})
+
+	children := make([]format.Node, 0, 10)
+	childLinks := make([]*format.Link, 0, 10)
+	for i := 0; i < 10; i++ {
+		child, _ := cbornode.WrapObject([]byte{byte(i)}, multihash.SHA2_256, -1)
+		children = append(children, child)
+		childLinks = append(childLinks, &format.Link{Name: fmt.Sprintf("child%d", i), Cid: child.Cid()})
+	}
+	b, err := cbornode.WrapObject(linker{Name: "root", Links: childLinks}, multihash.SHA2_256, -1)
+	if err != nil {
+		return "", fmt.Errorf("couldn't make cbor node: %v", err)
+	}
+	ds.Nodes[b.Cid()] = b
+
+	if err := car.WriteCar(context.Background(), &ds, []cid.Cid{b.Cid()}, f); err != nil {
+		return "", err
+	}
+
+	return f.Name(), nil
+}
+*/
+
+func TestIndexRT(t *testing.T) {
+	/*
+		carFile, err := mkCar()
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(carFile)
+	*/
+	carFile := "test.car"
```

Comment:
I think this will edit files in the package directory where the Go files are, meaning it could mess with the source code or leave files behind if the test panics/fails too early.

I'd use t.TempDir in all these that work with paths on the filesystem - feel free to leave as a TODO.

Reply:
Also, that file should be put in `v2/carbs/testdata/test.car`, because otherwise the Go module zip will include the file as part of the `v2/carbs` package source code